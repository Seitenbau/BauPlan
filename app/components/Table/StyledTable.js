import styled from 'styled-components';
import { rem, shadeColor } from 'utils/helper';

const Tabel = styled.div`
  background-color: #d3d3d3;
  position: absolute;
  box-sizing: border-box;
  height: ${(props) => rem((props.height / 2) * 100)};
  width: ${(props) => rem((props.width / 2) * 100)};
  border: 1px solid ${shadeColor('#d3d3d3', -0.8)};
  left: ${(props) => rem(props.left)};
  top: ${(props) => rem(props.top)};
  :after {
    content: "";
    display: block;
    position: absolute;
    border: 1px dashed ${shadeColor('#d3d3d3', -0.8)};
    border-top: none;
    height: 100%;
    width: 100%;
    top: ${(props) => rem((props.height / 2) * 100)};
  }
`;

export default Tabel;
