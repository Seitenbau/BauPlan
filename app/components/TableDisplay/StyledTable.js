import styled from 'styled-components';
import { rem, shadeColor } from 'utils/helper';

const Tabel = styled.div`
  background-color: #d3d3d3;
  position: absolute;
  box-sizing: border-box;
  height: ${(props) => rem(props.height * props.scaleFactor)};
  width: ${(props) => rem(props.width * props.scaleFactor)};
  border: 1px solid ${shadeColor('#d3d3d3', -0.8)};
  left: ${(props) => rem(props.left * props.scaleFactor)};
  top: ${(props) => rem(props.top * props.scaleFactor)};
  transform: rotate(${(props) => props.rotation}deg);
  :after {
    content: "";
    display: block;
    position: absolute;
    border: 1px dashed ${shadeColor('#d3d3d3', -0.8)};
    border-top: none;
    height: 100%;
    width: 100%;
    top: ${(props) => rem(props.height * props.scaleFactor)};
  }
  :hover {
    .projectIdentifier-reactangle {
      display: block;
    }
  }
`;

export default Tabel;
