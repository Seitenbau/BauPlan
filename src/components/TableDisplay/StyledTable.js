import styled from 'styled-components';
import { rem, shadeColor } from 'utils/helper';

// we use attrs for attributes which are often changing
const Table = styled.div.attrs({
  style: props => ({
    height: rem(props.height),
    width: rem(props.width),
    left: rem(props.left),
    top: rem(props.top)
  })
})`
  background-color: #d3d3d3;
  position: absolute;
  box-sizing: border-box;
  border: 1px solid ${shadeColor('#d3d3d3', -0.8)};
  transform: rotate(${props => props.rotation}deg);
  :after {
    content: '';
    display: block;
    position: absolute;
    border: 1px dashed ${shadeColor('#d3d3d3', -0.8)};
    border-top: none;
    height: 100%;
    width: 100%;
    top: 100%;
  }
  :hover {
    .projectIdentifier-reactangle {
      display: block;
    }
  }
`;

export default Table;
