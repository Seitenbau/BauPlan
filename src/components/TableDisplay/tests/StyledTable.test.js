import React from 'react';
import { shallow } from 'enzyme';

import StyledTable from '../StyledTable';

describe('<TableDisplay />', () => {
  const renderedComponent = shallow(<StyledTable height={80} width={40} left={100} top={100} rotation={90} />);
  it('Expect StyledTable to have styles defined', () => {
    expect(renderedComponent).toMatchSnapshot();
  });
});
