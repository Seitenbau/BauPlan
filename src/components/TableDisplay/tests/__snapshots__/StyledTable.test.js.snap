exports[`<TableDisplay /> Expect StyledTable to have styles defined 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="sc-bdVaJa iwhLko"
    height={80}
    width={40} />,
  "nodes": Array [
    <div
      className="sc-bdVaJa iwhLko"
      height={80}
      width={40} />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <styled.div
        height={80}
        left={100}
        rotation={90}
        top={100}
        width={40} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StyledComponent {
        "_reactInternalInstance": [Circular],
        "attrs": Object {},
        "context": Object {
          "__styled-components__": undefined,
        },
        "props": Object {
          "height": 80,
          "left": 100,
          "rotation": 90,
          "top": 100,
          "width": 40,
        },
        "refs": Object {},
        "state": Object {
          "generatedClassName": "iwhLko",
          "theme": Object {},
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="sc-bdVaJa iwhLko"
          height={80}
          width={40} />,
        "_debugID": 2,
        "_renderedOutput": <div
          className="sc-bdVaJa iwhLko"
          height={80}
          width={40} />,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <styled.div
    height={80}
    left={100}
    rotation={90}
    top={100}
    width={40} />,
}
`;
